infra_event_notifier/notifier.py

```diff
@@ -0,0 +1,40 @@
+from typing import Mapping
+
+from backends.datadog import notify_datadog
+from backends.jira import notify_jira
+from backends.slack import notify_slack
+
+
+def notify(
+    title: str,
+    text: str,
+    tags: Mapping[str, str],
+    datadog_api_key: str,
+    slack_api_key: str,
+    jira_api_key: str,
```

Comment:
consider making these optional?

i think it would be more flexible if this package allowed you to configure each of these notification targets independently, and still worked without all 3 being set up.

Reply:
i think a nicer API might be to define and load the API keys once on startup, rather than require the user to pass them along with every single call to `notify` which can be verbose

Reply:
you could use the sentry sdk as inspiration -- you only call `init` once with your API key, and after that you don't need to pass it to every single call to capture data https://github.com/getsentry/sentry-python

Reply:
What about a builder pattern?
That is designed for building objects that can be initialized in several ways.

```
notif = Notification(TITLE, ......)
   .withDatadog(DD_KEY .... anything DD specific)
   .withJira(JIRA_KEY, .....)
   # not adding slack because ....

notif.send()
```
That would avoid a proliferation of attributes on the send function call.

Or build the transport only, more in line with what Lyn suggests:
```
notif_transport = NotificationTransport()
   .withDatadog(DD_KEY .... anything DD specific)
   .withJira(JIRA_KEY, .....)
   # not adding slack because ....

notif_transport.send(.....)
....
notif_transport.send(.....) # Send another notification
```

Re the sentry-python `init`. I think it would be a good idea, though, please, avoid making it a global variable. You do not need that. It is done that way in the sdk to allow people capturing errors everywhere in their code without having to pass the sdk context around everywhere. 
You do not have that problem here so, please don't hesitate to pass around whatever you initialize.